api/src/main/java/org/pmiops/workbench/survey/NewUserSatisfactionSurveyServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.pmiops.workbench.survey;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import org.pmiops.workbench.access.AccessTierService;
+import org.pmiops.workbench.db.dao.UserAccessTierDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbUserAccessTier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NewUserSatisfactionSurveyServiceImpl implements NewUserSatisfactionSurveyService {
+
+  private final UserAccessTierDao userAccessTierDao;
+  private AccessTierService accessTierService;
+  private final Clock clock;
+
+  @Autowired
+  public NewUserSatisfactionSurveyServiceImpl(
+      UserAccessTierDao userAccessTierDao, AccessTierService accessTierService, Clock clock) {
+    this.userAccessTierDao = userAccessTierDao;
+    this.accessTierService = accessTierService;
+    this.clock = clock;
+  }
+
+  @Override
+  public boolean eligibleToTakeSurvey(DbUser user) {
+    if (user.getNewUserSatisfactionSurvey() != null) {
+      return false;
+    }
+
+    DbAccessTier registeredAccessTier = this.accessTierService.getRegisteredTierOrThrow();
+
+    Optional<DbUserAccessTier> userRegisteredAccessTier =
+        this.userAccessTierDao.getByUserAndAccessTier(user, registeredAccessTier);
+
+    if (!userRegisteredAccessTier.isPresent()) {
+      return false;
+    }
+
+    Instant enabledTime = userRegisteredAccessTier.get().getFirstEnabled().toInstant();
```

Comment:
stylistically, `Optional.get()` is discouraged.  Rather than `isPresent()` followed by `get()`, `ifPresent(entity -> ...)` is preferred.  (Tests can be an exception to this)

So I'd prefer the bulk of this method to look more like
```
boolean eligible = false;
userAccessTierDao.getByUserAndAccessTier(user, registeredAccessTier)
  .ifPresent(userTier -> {
    ...
    eligible = the date math comparison;
  });
return eligible;
```

Reply:
That makes sense, updated.

I used `.map(...).orElse(...)` instead because I got an error about using non-final variables in lambdas when using `ifPresent`.

In the controller, I have the line `DbUser user = this.userProvider.get();`. Should that be changed, or is it ok to trust that auth has handled user existence prior to that call?